### PR TITLE
Add a new doc describing userData and AMI support

### PR DESCRIPTION
**Pull request by suket22**
**1. Issue, if available:**
https://github.com/aws/karpenter/issues/885
https://github.com/aws/karpenter/issues/1008

**2. Description of changes:**
Adds a doc that describes how the support for UserData and AMI in the AWS Provider could look like. 

**3. How was this change tested?**
None, it's just a doc

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:36]: s/example-/example:/g

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:39]: It's probably a good idea to specify `toml` in here so that color coding can be done correctly.

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:54]: You can use the unicode `…` character instead of actual three dots

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:64]: I recommend using `foo/bar/baz` for test values.

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:111]: I think this can be improved to "leave it to Karpenter" or simply "let Karpenter"

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:115]: need a `.` at the end of this bullet point to stay consistent?

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:122]: "helps with"?

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:173]: I think you probably meant "backwards compatibility" not "backwards breaking compatibility"

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:183]: learning or leaning?

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:212]: "provisioner field what" to "provisioner field to specify what" ?

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:262]: All letters in CPU should be capitalized.

File: designs/aws/custom-user-data-and-amis.md
Comment (spring1843) [inline designs/aws/custom-user-data-and-amis.md:268]: There are a few double empty lines in the file, not sure if that serves any purpose.

**Review by spring1843 — DISMISSED** [review]
LGTM, there are some minor readability suggestions.

File: designs/aws/custom-user-data-and-amis.md
Comment (suket22) [reply designs/aws/custom-user-data-and-amis.md:268]: I don't think the formatting is something I'm really concerned with addressing before merging in. The double lines were just things I'd added in as line separators. 

Apologies for some of the typos, I'll get those fixed.

**Review by bwagner5 — COMMENTED** [review]
nice! Just a few nits, but otherwise, lgtm

**Review by bwagner5 — APPROVED** [review]
lgtm!

**Review by spring1843 — APPROVED** [review]
LGTM

**Pull request merged**
